### PR TITLE
feat: make entire tip resource area clickable

### PR DESCRIPTION
**Pull request by M4dhav**
fixed issue as mentioned in #86

### CODE REVIEW

File: lib/presentation/pages/tip_information_page.dart [hunk 1]
```diff
@@ -4,11 +4,18 @@ import 'package:infinite_horizons/presentation/atoms/atoms.dart';
 import 'package:infinite_horizons/presentation/molecules/molecules.dart';
 import 'package:url_launcher/url_launcher.dart';
 
-class TipInformationPage extends StatelessWidget {
-  const TipInformationPage(this.tip);
-
+class TipInformationPage extends StatefulWidget {
```
Comment (guyluz11) [inline lib/presentation/pages/tip_information_page.dart:7]: Why it got changed to StatefulWidget?, there is no reason for that.
Please revert this change

**Review by guyluz11 — CHANGES_REQUESTED** [review]

File: lib/presentation/pages/tip_information_page.dart (hunk 1 above)
Comment (M4dhav) [reply lib/presentation/pages/tip_information_page.dart:7]: My bad, I was trying another implementation earlier and forgot to revert this

File: lib/presentation/pages/tip_information_page.dart [hunk 2]
```diff
@@ -77,25 +79,35 @@ class TipInformationPage extends StatelessWidget {
...
-                                        onPressed: () {
-                                          if (link == null) {
-                                            return;
-                                          }
-                                          launchUrl(link);
-                                        },
-                                        icon: const Icon(Icons.link),
-                                      ),
-                                    ],
+                                  body: InkWell(
```
Comment (guyluz11) [inline lib/presentation/pages/tip_information_page.dart:87]: The body should open and close when clicked, not open the link

File: lib/presentation/pages/all_tips_page.dart [hunk 3]
```diff
@@ -27,7 +27,7 @@ class AllTipsPage extends StatelessWidget {
               trailing: const Icon(Icons.arrow_forward),
               onTap: () => Navigator.of(context).push(
                 MaterialPageRoute(
-                  builder: (context) => TipInformationPage(tip),
```
Comment (guyluz11) [inline lib/presentation/pages/all_tips_page.dart:30]: There is no reason for change

**Review by guyluz11 — CHANGES_REQUESTED** [review]

**Review by guyluz11 — CHANGES_REQUESTED** [review]
You missed a couple of comments

**Comment by M4dhav** [comment]
Could you please elaborate? I can't seem to find any comments added to the files in this PR.

**Comment by guyluz11** [comment]
> Could you please elaborate? I can't seem to find any comments added to the files in this PR.

image

.

image

**Comment by M4dhav** [comment]
image
I'm not sure why you're unable to see the replies but I have made them

**Comment by guyluz11** [comment]
Hey,
Can you fix the conflicts?

Thanks

File: lib/presentation/pages/tip_information_page.dart [hunk 4]
```diff
@@ -86,26 +88,35 @@ class TipInformationPage extends StatelessWidget {
...
-                                              onPressed: () {
-                                                if (link == null) {
-                                                  return;
-                                                }
-                                                launchUrl(link);
-                                              },
-                                              icon: const Icon(Icons.link),
-                                            ),
-                                          ],
+                                        body: InkWell(
```
Comment (guyluz11) [inline lib/presentation/pages/tip_information_page.dart:96]: The body should open and close when clicked, not open the link

**Review by guyluz11 — CHANGES_REQUESTED** [review]

**Comment by M4dhav** [comment]
I took the liberty of replacing Icon Button with Icon as InkWell makes the entire area clickable and makes Icon Button redundant. I also added Padding to the text to improve the look

**Comment by guyluz11** [comment]
Maybe I wasn't clear

Area number 1 only expends and retracts when clicked.
Area number 2 is the only play that opens the link.


image

**Comment by M4dhav** [comment]
I guess there was a communication gap, I will modify my implementation

**Comment by M4dhav** [comment]
Still working on this, I can't unrequest reviews so don't review it yet

**Review by guyluz11 — CHANGES_REQUESTED** [review]
"Still working on this, I can't unrequest reviews so don't review it yet"

Here you go

**Comment by M4dhav** [comment]
Alright, I'm done

File: lib/presentation/pages/tip_information_page.dart [hunk 5]
```diff
@@ -75,48 +94,72 @@ class TipInformationPage extends StatelessWidget {
...
-                                children:
-                                    tip.resourceLinks.map<ExpansionPanelRadio>(
+                              ExpansionPanelList(
+                                expansionCallback: (panelIndex, expanded) {
+                                  setState(() {
+                                    isExpanded.forEach((key, value) {
+                                      isExpanded[key] = false;
+                                    });
+                                    isExpanded[isExpanded.keys
+                                        .elementAt(panelIndex)] = expanded;
```
Comment (guyluz11) [inline lib/presentation/pages/tip_information_page.dart:104]: I don't like this code, it does not look clean.

**Review by guyluz11 — APPROVED** [review]
